File: nodejs-examples/test-offline-qwen3-asr.js
// Copyright (c)  2026  Xiaomi Corporation (authors: Fangjun Kuang)
//
const sherpa_onnx = require('sherpa-onnx');

function createOfflineRecognizer() {
  const config = {
    modelConfig: {
      qwen3Asr: {
        convFrontend:
            './sherpa-onnx-qwen3-asr-0.6B-int8-2026-03-25/conv_frontend.onnx',
        encoder:
            './sherpa-onnx-qwen3-asr-0.6B-int8-2026-03-25/encoder.int8.onnx',
        decoder:
            './sherpa-onnx-qwen3-asr-0.6B-int8-2026-03-25/decoder.int8.onnx',
        tokenizer: './sherpa-onnx-qwen3-asr-0.6B-int8-2026-03-25/tokenizer',
        hotwords: '',
      },
      tokens: '',
    }
  };

  return sherpa_onnx.createOfflineRecognizer(config);
}

const recognizer = createOfflineRecognizer();
const stream = recognizer.createStream();

const waveFilename =
    './sherpa-onnx-qwen3-asr-0.6B-int8-2026-03-25/test_wavs/raokouling.wav';
const wave = sherpa_onnx.readWave(waveFilename);
stream.acceptWaveform(wave.sampleRate, wave.samples);

recognizer.decode(stream);
const text = recognizer.getResult(stream).text;
console.log(text);

stream.free();
recognizer.free();

File: kotlin-api-examples/test_offline_qwen3_asr.kt
package com.k2fsa.sherpa.onnx

fun main() {
  val recognizer = createOfflineRecognizer()
  val waveFilename = "./sherpa-onnx-qwen3-asr-0.6B-int8-2026-03-25/test_wavs/raokouling.wav"

  val waveData = WaveReader.readWaveFromFile(
      filename = waveFilename,
  )

  var stream = recognizer.createStream()
  stream.acceptWaveform(waveData.samples, sampleRate=waveData.sampleRate)

  val start = System.currentTimeMillis()
  recognizer.decode(stream)
  val stop = System.currentTimeMillis()

  var result = recognizer.getResult(stream)
  println(result)

  val timeElapsedSeconds = (stop - start) / 1000.0f
  val audioDuration = waveData.samples.size / waveData.sampleRate.toFloat()
  val realTimeFactor = timeElapsedSeconds / audioDuration

  println(String.format("-- elapsed : %.3f seconds", timeElapsedSeconds))
  println(String.format("-- audio duration: %.3f seconds", audioDuration))
  println(String.format("-- real-time factor (RTF): %.3f", realTimeFactor))

  stream.release()
  recognizer.release()
}


fun createOfflineRecognizer(): OfflineRecognizer {
  val config = OfflineRecognizerConfig(
      modelConfig = getOfflineModelConfig(type = 61)!!,
  )

  return OfflineRecognizer(config = config)
}
